drill-shaded/guava-shaded/pom.xml

```diff
@@ -0,0 +1,86 @@
+<?xml version="1.0"?>
+<!--
+
+    Licensed to the Apache Software Foundation (ASF) under one
+    or more contributor license agreements.  See the NOTICE file
+    distributed with this work for additional information
+    regarding copyright ownership.  The ASF licenses this file
+    to you under the Apache License, Version 2.0 (the
+    "License"); you may not use this file except in compliance
+    with the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <parent>
+    <groupId>org.apache.drill</groupId>
+    <artifactId>drill-shaded</artifactId>
+    <version>1.0</version>
+  </parent>
+
+  <artifactId>guava-shaded</artifactId>
+  <version>23.0</version>
+  <name>drill-shaded/guava-shaded</name>
+
+  <properties>
+    <!-- skip source archives -->
+    <assembly.attach>false</assembly.attach>
+    <dep.guava.version>23.0</dep.guava.version>
+  </properties>
+
+  <dependencies>
+    <dependency>
+      <groupId>com.google.guava</groupId>
+      <artifactId>guava</artifactId>
+      <version>${dep.guava.version}</version>
+    </dependency>
+    <dependency>
+      <groupId>com.google.errorprone</groupId>
+      <artifactId>error_prone_annotations</artifactId>
+      <version>2.0.18</version>
+    </dependency>
+  </dependencies>
+
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-shade-plugin</artifactId>
+        <version>3.1.0</version>
+        <executions>
+          <execution>
+            <phase>package</phase>
+            <goals>
+              <goal>shade</goal>
+            </goals>
+            <configuration>
+              <createSourcesJar>true</createSourcesJar>
```

Comment:
`shadeSourcesContent`?

Reply:
I don't understand and/or see the problem.

Reply:
If we have both `createSourcesJar` and `shadeSourcesContent` set to true and handle `com.google` pattern in relocation, all imports that match this pattern will be also relocated for source files, even if those classes don't belong to `guava` lib.

For example, with such config, in the source for `Preconditions` will be available import `org.apache.drill.shaded.guava.com.google.errorprone.annotations.CanIgnoreReturnValue`, but class `CanIgnoreReturnValue` was in `error_prone_annotations` jar, not in `guava`.

Reply:
I see, but it is actually not a problem as those sources are provided for debugging, not for compiling shaded jars. Note that maven default Shader simply does regular expression replacement from `pattern` to `shadedPattern` when `shadeSourcesContent` is `true`. Please add a comment explaining why two patterns are used or use a single common pattern.